Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -949,10 +987,14 @@ void HTMLInputElement::reset_algorithm()
 
     // and then invoke the value sanitization algorithm, if the type attribute's current state defines one.
     m_value = value_sanitization_algorithm(m_value);
+
     if (m_text_node) {
         m_text_node->set_data(MUST(String::from_deprecated_string(m_value)));
         update_placeholder_visibility();
     }
+
+    if (type_state() == TypeAttributeState::Color)
+        MUST(m_color_well_element->style_for_bindings()->set_property(CSS::PropertyID::BackgroundColor, m_value));
```

Comment:
I... don't know if `m_color_well_element` could be null here but it should be fine to check for that anyway.